### DISCUSSION TITLE
How to observe just viewStore.send with WithViewStore?

### DISCUSSION THREAD

**Discussion by theMuslimDev** [pointfreeco/swift-composable-architecture#2119] [Q&A] [answered]
If we want to observe a small piece of state inside a view. We would do something like this:

```swift
struct SomeView: some View {
let store: StoreOf<SomeLogic>
  var body: some View {
    WithViewStore(self.store, observe: \.someState) { viewStore in
      Text(viewStore.someState)
    }
  }
}
```

How can I achieve a similar scenario where I only need to observe sending an action? Will I just use `{$0}`?

**Comment by theMuslimDev** [comment, marked as answer]
This answer by @tgrapperon answers this.

https://github.com/pointfreeco/swift-composable-architecture/discussions/1666#discussioncomment-4141780

For people that don't want to open the link. In instance where you are just sending an action and not displaying some information in the UI, you can create a ViewStore on the spot when you send the action:

```swift
Button("Send") { ViewStore(store.stateless).send(.someAction) }
```

**Reply by tgrapperon** [reply]
I saw your message earlier today and I was planning to respond later! Fortunately, past-myself beat me to it! I'm seizing the occasion to also remind that you don't need to observe anything to scope a store. See https://github.com/pointfreeco/swift-composable-architecture/discussions/1666#discussioncomment-4141780.

**Reply by theMuslimDev** [reply]
Nice. That's the exact comment I referenced in the answer lol

**Reply by tgrapperon** [reply]
Damn current-myself is not as awake as past-myself!
